sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -859,24 +863,33 @@ def update_repository_properties(
         )
 
     @distributed_trace
-    def upload_manifest(
-        self, repository: str, manifest: Union[OCIManifest, IO], *, tag: Optional[str] = None, **kwargs
+    def set_manifest(
+        self,
+        repository: str,
+        manifest: Union[JSON, IO[bytes]],
+        *,
+        tag: Optional[str] = None,
+        media_type: str = OCI_IMAGE_MANIFEST,
+        **kwargs
     ) -> str:
-        """Upload a manifest for an OCI artifact.
+        """Set a manifest for an artifact.
 
-        :param str repository: Name of the repository.
-        :param manifest: The manifest to upload. Note: This must be a seekable stream.
-        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :param str repository: Name of the repository
+        :param manifest: The manifest to set. It can be a JSON formatted dict or seekable stream.
+        :type manifest: dict or IO
         :keyword tag: Tag of the manifest.
         :paramtype tag: str or None
-        :returns: The digest of the uploaded manifest, calculated by the registry.
+        :keyword media_type: The media type of the manifest. If not specified, this value will be set to
+            a default value of "application/vnd.oci.image.manifest.v1+json".
+        :paramtype media_type: str
+        :returns: The digest of the set manifest, calculated by the registry.
         :rtype: str
         :raises ValueError: If the parameter repository or manifest is None,
-            or the digest in the response does not match the digest of the uploaded manifest.
+            or the digest in the response does not match the digest of the set manifest.
         """
         try:
-            if isinstance(manifest, OCIManifest):
-                data = _serialize_manifest(manifest)
+            if isinstance(manifest, MutableMapping):
+                data: IO[bytes] = BytesIO(_serialize_manifest(manifest))
```

Comment:
We shouldn't need to do this - what happens if you just pass it through?

Reply:
Yeah, I also thought it's not necessary. If just pass it, will meet mypy error like this: 
![bdb90d52-ff5c-4c7c-95cb-b6ae9912c234](https://user-images.githubusercontent.com/20915761/232618082-07cdfa72-f902-4c00-a85d-09e8325a0f8b.jpg)
cc: @kristapratico Any other way to fix the mypy error?

Reply:
Sorry - which one is the mypy error if you just pass it through?

Reply:
ping @YalinLi0312 - would be good to get this resolved this week.
I'm guessing it's the last line `no attribute "read" error`
I see that this API is being generated as a stream - and I'm not sure why and it's not really a data streaming scenario.

Reply:
I think we see this because of the change we made in _helpers in this PR:

![image](https://user-images.githubusercontent.com/31998003/233213014-27af0c76-8123-41e4-b8aa-87cef476aa7f.png)

Looks like the generated code is expecting `IO[Any]` and we're passing `bytes`

Reply:
This is the error when I pass it through: 
```text
azure/containerregistry/_container_registry_client.py:894:` error: Incompatible types in assignment (expression has type "IO[bytes]", variable has type "BytesIO")  [assignment]
```

Reply:
We still need to fix this - but I don't want to block this PR on it - could we please open an issue to address this separately?

Reply:
Ignored the mypy error here, and logged an issue: https://github.com/Azure/azure-sdk-for-python/issues/30053

Reply:
Update: fixed the mypy error from a recommendation by @kristapratico in my last commit.